src/circuit.rs

```diff
@@ -1441,4 +1445,232 @@ mod tests {
             .render(K, &circuit, &root)
             .unwrap();
     }
+
+    fn generate_circuit_instance<R: RngCore>(
+        is_native_asset: bool,
+        split_flag: bool,
+        mut rng: R,
+    ) -> (Circuit, Instance) {
+        // Create asset
+        let asset_base = if is_native_asset {
+            AssetBase::native()
+        } else {
+            AssetBase::random(&mut rng)
+        };
+
+        // Create spent_note
+        let (spent_note_fvk, spent_note) = {
+            let sk = SpendingKey::random(&mut rng);
+            let fvk: FullViewingKey = (&sk).into();
+            let sender_address = fvk.address_at(0u32, Scope::External);
+            let rho_old = Nullifier::dummy(&mut rng);
+            let spent_note = Note::new(
+                sender_address,
+                NoteValue::from_raw(40),
+                asset_base,
+                rho_old,
+                &mut rng,
+            );
+            (fvk, spent_note)
+        };
+
+        let (dummy_sk, fvk, scope, nf_old) = if split_flag {
+            let sk = SpendingKey::random(&mut rng);
+            let fvk: FullViewingKey = (&sk).into();
+            (
+                Some(sk),
+                fvk.clone(),
+                Scope::External,
+                spent_note.nullifier(&fvk),
+            )
+        } else {
+            (
+                None,
+                spent_note_fvk.clone(),
+                spent_note_fvk
+                    .scope_for_address(&spent_note.recipient())
+                    .unwrap(),
+                spent_note.nullifier(&spent_note_fvk),
+            )
+        };
+        let ak: SpendValidatingKey = fvk.clone().into();
+        let alpha = pallas::Scalar::random(&mut rng);
+        let rk = ak.randomize(&alpha);
+
+        let output_note = {
+            let sk = SpendingKey::random(&mut rng);
+            let fvk: FullViewingKey = (&sk).into();
+            let sender_address = fvk.address_at(0u32, Scope::External);
+
+            Note::new(
+                sender_address,
+                NoteValue::from_raw(10),
+                asset_base,
+                nf_old,
+                &mut rng,
+            )
+        };
+
+        let cmx = output_note.commitment().into();
+
+        let value = if split_flag {
+            NoteValue::zero() - output_note.value()
+        } else {
+            spent_note.value() - output_note.value()
+        };
```

Comment:
Let's unify this part with the previous `if split_flag`. Plus a comment to describe rationale.

Reply:
Done